### PR TITLE
reconcile pyxis api secret

### PR DESCRIPTION
**Pull request by skattoju**
Addresses #9 

The pyxis api secret name can optionally be configured via the cr

This is going to have a bunch of repeated code to be addressed in #24

### CODE REVIEW

File: controllers/secret.go [hunk 1]
```diff
@@ -96,18 +96,47 @@ func (r *OperatorPipelineReconciler) reconcileGitHubAPISecret(meta metav1.Object
...
-
-	return errors.New("pyxis api secret not found")
+	log.Info(fmt.Sprintf("existing %s secret found in namespace %s", secretName, meta.Namespace))
+	kubeConfigSecret := &corev1.Secret{}
+	err = r.Client.Get(context.TODO(), key, kubeConfigSecret)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to get the %s secret", secretName))
+		return err
+	}
+	if kubeConfigSecret.Data[KUBECONFIG_SECRET] == nil {
```
Comment (acornett21) [inline controllers/secret.go:134]: This references the `kubeconfig` value. the value for this key should be
`PYXIS_API_KEY     = "pyxis_api_key"`

Also. checking in a map this way can throw a panic if the key isn't present.
```
if value, ok := kubeConfigSecret.Data[KUBECONFIG_SECRET]; ok{
//then evaluate data in value
}
```

Comment (skattoju) [reply controllers/secret.go:134]: I think if the requested key doesn’t exist, we get the value type’s zero value..

File: controllers/secret.go [hunk 2]
```diff
@@ -33,19 +35,47 @@ const (
...
-		Name:      KUBECONFIG_SECRET,
+		Name:      secretName,
 	}
-
 	secret := newSecret(key)
-	if IsObjectFound(r.Client, key, secret) {
-		log.Info("existing kubeconfig secret found")
-		return nil // Existing Secret found, do nothing...
+	if !IsObjectFound(r.Client, key, secret) {
+		err := errors.New(fmt.Sprintf("An existing secret named %s was not found in namespace %s", secretName, meta.Namespace))
```
[the remark this answers is not in the document]
Comment (skattoju) [reply controllers/secret.go]: agreed, should be fixed in #24

File: controllers/secret.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (skattoju) [reply controllers/secret.go:134]: added an || len == 0

Comment (skattoju) [reply controllers/secret.go:134]: Missed that was the same as the secret name in the other two cases.. i think.

File: controllers/secret.go [hunk 3]
```diff
@@ -33,19 +35,47 @@ const (
...
+		log.Error(err, fmt.Sprintf("unable to reconcile kubeconfig in namespace %s", meta.Namespace))
+		return err
 	}
-
-	log.Info("creating new kubeconfig secret")
-	return r.Client.Create(context.TODO(), secret)
+	log.Info(fmt.Sprintf("existing %s secret found in namespace %s", secretName, meta.Namespace))
+	kubeConfigSecret := &corev1.Secret{}
+	err = r.Client.Get(context.TODO(), key, kubeConfigSecret)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (skattoju) [reply controllers/secret.go]: merged 👍

File: controllers/secret.go [hunk 4]
```diff
@@ -33,19 +35,47 @@ const (
...
+		log.Error(err, fmt.Sprintf("unable to get the %s secret", secretName))
```
[the remark this answers is not in the document]
Comment (skattoju) [reply controllers/secret.go]: its logged right before as an info but I agree its better to include it..

have not looked into into completely but apparently the k8s runtime logging interface should be able add contextual info like namespace etc to the logs 🤔

**Review by samira-barouti — APPROVED** [review]
/lgtm

**Pull request merged**